google/ggrc-core
# GGRC-3519 Fetch Related Assessments based on Assessment Type

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] #6765

# Issue description

Related assessments fetching should be based on assessment type.

# Steps to test the changes

1. Create several Assessments of a Control type and map several controls to them.
2. Map some other objects ( Objectives, Risks, etc.  ) to the just created assessments.
3. Open an info-pane of one of the assessments and go to the Related assessments tab.
4. 5th column of the table should be named like 'Related {assessment_type}' and should only include mapped objects of the assessment type ( previously it include all mapped objects  )

6. Open one of controls you just mapped to the assessments. Open Related Assessments popup from the control's info-pane.
7. 5th column of the table should be named like 'Related {object_type}' and the table should only include assessments which have connection through this control.

Please see the schema in the JIRA ticket.

# Solution description

This PR dynamically displays the correct title for column with the related objects in the related assessments table and adjusts Query API request to fetch assessment only connected with the objects of the assessment type ( previously related assessment where loaded based on the connection through mapped Controls and Objectives ).

For the assessments related to some an object we use the 'similar' filter provided by the backend. This filter is available for Assessment, Control and Objective objects.

# Sanity checklist


# PR Review checklist
